src/components/PopupWithForm.js

```diff
@@ -1,36 +1,45 @@
-import PopUp from "./PopUp.js";
+import PopUp from "./Popup.js";
 
 export default class PopUpWithForm extends PopUp {
-  constructor(popupSelector, handleFormSubmit) {
-    super({ popupSelector });
-    this._popupForm = this._popupElement.querySelector(".modal__form");
+  constructor(popUpSelector, handleFormSubmit) {
+    super({ popUpSelector });
+    this._popUpForm = this._popUpElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputEls = Array.from(
-      this._popupForm.querySelectorAll(".modal__input")
-    );
-  }
+      this._popUpForm.querySelectorAll(".modal__input"));
+    this._submitBtn = this._popUpForm.querySelector(".modal__button");
+      }
+  
 
   close() {
-    this._popupForm.reset();
+    this._popUpForm.reset();
     super.close();
   }
 
   _getInputValues() {
-    const formValues = [];
+    const formValues = {};
     this._inputEls.forEach((element) => {
       formValues[element.name] = element.value;
     });
 
     return formValues;
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitBtn.textContent = "Saving";
```

Comment:
` "Saving"` should be with the 3 dots  ` "Saving..."` to clearly show the loading effect